inference/worker/work.py

```diff
@@ -106,9 +107,25 @@ def handle_work_request(
     work_request: inference.WorkRequest,
     worker_config: inference.WorkerConfig,
 ):
-    prompt, parameters = make_prompt_and_parameters(tokenizer=tokenizer, work_request=work_request)
+    parameters = interface.GenerateStreamParameters.from_work_parameters(work_request.parameters)
+    prompt = ""
+    used_plugin = None
+
+    # prompt, parameters = make_prompt_and_parameters(tokenizer=tokenizer, work_request=work_request)
+
+    prompt, used_plugin = chat_chain.handle_conversation(work_request, worker_config, parameters, tokenizer)
     logger.debug(f"Prompt: {prompt}")
 
+    # If plugin was used, lockup sampling parameters to "best" known values for plugins
+    # and llama sft-7e3 model for now...
+    if used_plugin:
```

Comment:
if I understand this correctly, this is for the final inference? i.e. after the plugin has been used? if yes, I wouldn't do this. I'd make a new entry (something like "k50-plugins", e.g. see a current open PR to add more of these) in the sampling parameter configs we give to the frontend and if we really want to "force" it, then the frontend should do it

Reply:
Yes that is correct, but if it gets only the final prompt. if sampling params are anything other than that, it just produces bad results for current sft models.

So we can lock them from the UI part, sound good to me.

Reply:
sounds good. I suggest upon PR merge, you create an issue for frontend. you can assign it to me